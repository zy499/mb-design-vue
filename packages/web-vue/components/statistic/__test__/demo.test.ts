import demoTest from '../../../scripts/demo-test';

demoTest('statistic');
