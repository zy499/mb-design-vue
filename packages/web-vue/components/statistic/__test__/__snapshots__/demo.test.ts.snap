// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<statistic> demo: render [animation] correctly 1`] = `
"<div class=\\"mb-statistic\\">
  <div class=\\"mb-statistic-title\\">User Growth Rate</div>
  <div class=\\"mb-statistic-content\\">
    <div class=\\"mb-statistic-value\\"><span class=\\"mb-statistic-prefix\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-arrow-rise\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M23.992 6.01a.01.01 0 0 1 .016 0l9.978 11.974a.01.01 0 0 1-.007.016H14.02a.01.01 0 0 1-.007-.016l9.978-11.975Z\\"></path><path d=\\"m24 6 10 12H14L24 6Z\\" fill=\\"currentColor\\" stroke=\\"none\\"></path><path d=\\"M26 42h-4V16h4z\\"></path><path fill=\\"currentColor\\" stroke=\\"none\\" d=\\"M26 42h-4V16h4z\\"></path></svg></span><span class=\\"mb-statistic-value-integer\\">0</span><span class=\\"mb-statistic-value-decimal\\"> .00</span><span class=\\"mb-statistic-suffix\\">%</span></div>
    <!--v-if-->
  </div>
</div><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
  <!--v-if-->Start
</button>"
`;

exports[`<statistic> demo: render [basic] correctly 1`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 24px;\\">
    <div class=\\"mb-statistic\\">
      <div class=\\"mb-statistic-title\\">Downloads</div>
      <div class=\\"mb-statistic-content\\">
        <div class=\\"mb-statistic-value\\">
          <!--v-if--><span class=\\"mb-statistic-value-integer\\">125,670</span>
          <!--v-if-->
          <!--v-if-->
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
  <div class=\\"mb-space-item\\">
    <div class=\\"mb-statistic\\">
      <!--v-if-->
      <div class=\\"mb-statistic-content\\">
        <div class=\\"mb-statistic-value\\">
          <!--v-if--><span class=\\"mb-statistic-value-integer\\">40509</span><span class=\\"mb-statistic-value-decimal\\"> .00</span>
          <!--v-if-->
        </div>
        <div class=\\"mb-statistic-extra\\">Comments</div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`<statistic> demo: render [countdown] correctly 1`] = `
"<div class=\\"mb-space mb-space-vertical\\">
  <div class=\\"mb-space-item\\" style=\\"margin-bottom: 8px;\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
      <!--v-if-->Start
    </button></div>
  <div class=\\"mb-space-item\\">
    <div class=\\"mb-statistic mb-statistic-countdown\\">
      <div class=\\"mb-statistic-title\\">Countdown</div>
      <div class=\\"mb-statistic-content\\">
        <div class=\\"mb-statistic-value\\">02:00:00</div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`<statistic> demo: render [prefix] correctly 1`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 24px;\\">
    <div class=\\"mb-statistic\\">
      <div class=\\"mb-statistic-title\\">New Users</div>
      <div class=\\"mb-statistic-content\\">
        <div class=\\"mb-statistic-value\\">
          <!--v-if--><span class=\\"mb-statistic-value-integer\\">125,670</span>
          <!--v-if--><span class=\\"mb-statistic-suffix\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-arrow-rise\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M23.992 6.01a.01.01 0 0 1 .016 0l9.978 11.974a.01.01 0 0 1-.007.016H14.02a.01.01 0 0 1-.007-.016l9.978-11.975Z\\"></path><path d=\\"m24 6 10 12H14L24 6Z\\" fill=\\"currentColor\\" stroke=\\"none\\"></path><path d=\\"M26 42h-4V16h4z\\"></path><path fill=\\"currentColor\\" stroke=\\"none\\" d=\\"M26 42h-4V16h4z\\"></path></svg></span>
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
  <div class=\\"mb-space-item\\">
    <div class=\\"mb-statistic\\">
      <div class=\\"mb-statistic-title\\">User Growth Rate</div>
      <div class=\\"mb-statistic-content\\">
        <div class=\\"mb-statistic-value\\"><span class=\\"mb-statistic-prefix\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-arrow-rise\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M23.992 6.01a.01.01 0 0 1 .016 0l9.978 11.974a.01.01 0 0 1-.007.016H14.02a.01.01 0 0 1-.007-.016l9.978-11.975Z\\"></path><path d=\\"m24 6 10 12H14L24 6Z\\" fill=\\"currentColor\\" stroke=\\"none\\"></path><path d=\\"M26 42h-4V16h4z\\"></path><path fill=\\"currentColor\\" stroke=\\"none\\" d=\\"M26 42h-4V16h4z\\"></path></svg></span><span class=\\"mb-statistic-value-integer\\">50</span><span class=\\"mb-statistic-value-decimal\\"> .52</span><span class=\\"mb-statistic-suffix\\">%</span></div>
        <!--v-if-->
      </div>
    </div>
  </div>
</div>"
`;
